frontend/src/components/activePage/TableInstance/RowInstanceActions/RowInstanceActionsExtended.tsx

```diff
@@ -3,13 +3,27 @@ import { Popover, Tooltip, Upload } from 'antd';
 import Button from 'antd-button-color';
 import { InfoOutlined, FolderOpenOutlined } from '@ant-design/icons';
 import { VmStatus } from '../../../../utils';
-import { ISSHInfo } from '../SSHModalContent/SSHModalContent';
+import { EnvironmentType } from '../../../../generated-types';
+
+const getSSHTooltipText = (
+  hasSSHKeys: boolean,
+  isInstanceReady: boolean,
+  environmentType: EnvironmentType
+) => {
+  if (environmentType === EnvironmentType.Container)
+    return 'Containers does not support SSH connection (yet!)';
+  if (!hasSSHKeys)
+    return 'You have no SSH keys associated with your account, go to Account page and add them';
+  if (!isInstanceReady) return 'Your instance is not ready yet, please wait';
+  return '';
```

Comment:
```suggestion
  return 'Show SSH connection instructions';
```

Reply:
Also, `Your instance is not ready yet, please wait` should be more like: `Instance must be ready in order to connect`